src/Features/Onboarding.php

```diff
@@ -921,7 +922,8 @@ public static function add_help_tab() {
 			$task_list_hidden = get_option( 'woocommerce_task_list_hidden', 'no' );
 			$onboarding_data  = get_option( self::PROFILE_DATA_OPTION, array() );
 			$is_completed     = isset( $onboarding_data['completed'] ) && true === $onboarding_data['completed'];
-			$is_enabled       = ! $is_completed;
+			$is_skipped       = isset( $onboarding_data['skipped'] ) && true === $onboarding_data['skipped'];
+			$is_enabled       = ! $is_completed && ! $is_skipped;
```

Comment:
It looks like we use this logic several times throughout the code (`! profilerCompleted && ! profilerSkipped`).  It would be great if we could somehow consolidate this logic into a single place so it's less error prone if we need to change it again.

I've also been thinking about how this works with being able to access the profiler after completion/skip and I see a couple potential ways we can improve the existing setup here:

1. Introduce a setting `isProfilerEnabled` from the server that contains this logic and allows us to toggle on/off if we need to override the logic to access the profiler.

2. Always allow the profiler to be accessed through its own route (or if some query param like `step` is defined) and let the UI redirect to the profiler if not complete or skipped.

I'm leaning towards option 2 because this gives us a lot of flexibility and would simplify issues like hiding the UI during connection (exposed in https://github.com/woocommerce/woocommerce-admin/pull/4717), but would love more feedback on this idea.

Reply:
At first glance option 2 makes a lot of sense and seems to provide opportunity to clean up a lot of conditional logic we have - I'm thinking about the "home" route render and the forced postback currently required to revisit the OBW from the home screen. Perhaps the only gotchas might be changing existing Tracks events?

Reply:
Thanks for your insights on this, Jeff!

>I'm thinking about the "home" route render and the forced postback currently required to revisit the OBW from the home screen.

This, the soon-to-be task to revisit the profiler, and oddities behind the connection flow make it seem like the right decision to me as well.

>Perhaps the only gotchas might be changing existing Tracks events?

Good question.  I think the main one that might be `storeprofiler_toggled` (`wcadmin_storeprofiler_toggled`) which would no longer be necessary.  Any others you see @jeffstieler?

@pmcpinto are we currently using this or is there any information you think we might miss out on from this event?  I imagine we could still determine if a user revisits after completion with the other existing events.

Reply:
Thank you @joshuatf and @jeffstieler for your suggestions, I made the changes you mentioned in the commit 74f344f.
As soon as Pedro confirms the things related to the event recording I will add the necessary modifications.

@joshuatf could you give another look at this PR?